client/src/components/geneExpression/menus/createGenesetDialogue.js

```diff
@@ -0,0 +1,146 @@
+import _ from "lodash";
+import React from "react";
+import { connect } from "react-redux";
+import AnnoDialog from "../../categorical/annoDialog";
+import LabelInput from "../../categorical/labelInput";
+// import LabelInput from "../labelInput";
+// import { genesetPrompt, isGenesetErroneous } from "../genesetUtil";
+
+@connect((state) => ({
+  annotations: state.annotations,
+  schema: state.annoMatrix?.schema,
+  ontology: state.ontology,
+  obsCrossfilter: state.obsCrossfilter,
+  genesetsUI: state.genesetsUI,
+}))
+class CreateGenesetDialogue extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      genesetName: "",
+      genesToPopulateGeneset: "",
+    };
+  }
+
+  disableCreateGenesetMode = (e) => {
+    const { dispatch } = this.props;
+    this.setState({
+      genesetName: "",
+      genesToPopulateGeneset: "",
+    });
+    dispatch({
+      type: "geneset: disable create geneset mode",
+    });
+    if (e) e.preventDefault();
+  };
+
+  createGeneset = (e) => {
+    const { dispatch } = this.props;
+    const { genesetName, genesToPopulateGeneset } = this.state;
+
+    dispatch({
+      type: "geneset: create",
+      genesetName,
+    });
+    if (genesToPopulateGeneset) {
+      const genesTmpHardcodedFormat = [];
+
+      const genesArrayFromString = _.pull(
+        _.uniq(genesToPopulateGeneset.split(/[ ,]+/)),
+        ""
+      );
+
+      genesArrayFromString.forEach((_gene) => {
+        genesTmpHardcodedFormat.push({
+          geneSymbol: _gene,
+        });
+      });
+
+      dispatch({
+        type: "geneset: add genes",
+        genesetName,
+        genes: genesTmpHardcodedFormat,
+      });
+    }
+    dispatch({
+      type: "geneset: disable create geneset mode",
+    });
+    this.setState({
+      genesetName: "",
+      genesToPopulateGeneset: "",
+    });
+    e.preventDefault();
+  };
+
+  genesetNameError = () => {
+    /* todo genesets validation */
```

Comment:
Is this going to be done in a later PR? Is the issue tracked somewhere? If so, mind adding the issue number here in the comments so that we have a paper trail that this will be implemented in a subsequent PR?

Reply:
cc @signechambers1

Reply:
Validation of names should be super trivial now that we decided to allow (more or less) any unicode character.   The backend has not yet accommodated this, but I think validation should be in the front-end.  Primarily it is simple stuff like:
* no duplicate genes in a gene set
* gene set name is unique
* gene set name does not include running or leading/trailing spaces (at least I think this is the new requirement)
So, TL;DR - very easy to add to this PR.

Reply:
@signechambers1 is creating an issue for validation